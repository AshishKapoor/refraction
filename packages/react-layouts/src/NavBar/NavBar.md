```js
const { MemoryRouter } = require('react-router-dom');
<MemoryRouter>
  <NavBar
    logo="Logo"
    breadcrumb="Breadcrumb"
    tools="Tools"
    title="Hello world"
  />
</MemoryRouter>;
```
