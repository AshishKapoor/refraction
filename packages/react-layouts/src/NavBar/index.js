/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import * as React from 'react';
import styled from '@emotion/styled/macro';
import { withFallback as wf } from '@quid/theme';

type Props = {
  logo?: React.Node,
  breadcrumb?: React.Node,
  tools?: React.Node,
};

const PADDING = wf(({ theme }) => theme.sizes.small);

const Left = styled.div`
  padding-top: ${PADDING};
  padding-left: ${PADDING};
  padding-bottom: ${PADDING};
`;

const Middle = styled.div`
  padding-left: calc(${PADDING} * 3);
`;

const Right = styled.div`
  margin-left: auto;
  padding-left: calc(${PADDING} * 3);
  padding-right: ${PADDING};
`;

const NavBar = styled(({ logo, breadcrumb, tools, ...props }: Props) => (
  <nav {...props}>
    <Left>{logo}</Left>
    <Middle>{breadcrumb}</Middle>
    <Right>{tools}</Right>
  </nav>
))`
  display: flex;
  flex-direction: row;
  height: 50px;
  align-items: center;
`;

export default NavBar;
