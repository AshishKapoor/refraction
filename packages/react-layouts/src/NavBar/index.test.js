/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import React from 'react';
import { mount } from 'enzyme';
import NavBar from './index';

it('renders the expected markup', () => {
  const wrapper = mount(
    <NavBar
      logo={<span>foo</span>}
      breadcrumb={<span>foo</span>}
      tools={<span>foo</span>}
      title={'Example title for the nav tag.'}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
