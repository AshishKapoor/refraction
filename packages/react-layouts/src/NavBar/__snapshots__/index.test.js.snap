// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected markup 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 50px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  padding-top: 5px;
  padding-left: 5px;
  padding-bottom: 5px;
}

.emotion-2 {
  padding-left: calc(5px * 3);
}

.emotion-4 {
  margin-left: auto;
  padding-left: calc(5px * 3);
  padding-right: 5px;
}

<NavBar
  breadcrumb={
    <span>
      foo
    </span>
  }
  logo={
    <span>
      foo
    </span>
  }
  title="Example title for the nav tag."
  tools={
    <span>
      foo
    </span>
  }
>
  <nav
    className="emotion-6 emotion-7"
    title="Example title for the nav tag."
  >
    <Left>
      <div
        className="emotion-0 emotion-1"
      >
        <span>
          foo
        </span>
      </div>
    </Left>
    <Middle>
      <div
        className="emotion-2 emotion-3"
      >
        <span>
          foo
        </span>
      </div>
    </Middle>
    <Right>
      <div
        className="emotion-4 emotion-5"
      >
        <span>
          foo
        </span>
      </div>
    </Right>
  </nav>
</NavBar>
`;
